scripts/sql/000003_add_endpoint_availability.up.sql

```diff
@@ -0,0 +1,77 @@
+BEGIN;
+
+CREATE TABLE IF NOT EXISTS fhir_endpoints_availability (
+    url             VARCHAR(500),
+    http_200_count       BIGINT,
+    http_all_count       BIGINT,
+    created_at                TIMESTAMPTZ NOT NULL DEFAULT NOW(),
+    updated_at                TIMESTAMPTZ NOT NULL DEFAULT NOW()
+);
+
+ALTER TABLE fhir_endpoints_info ADD COLUMN availability DECIMAL(64,4);
+ALTER TABLE fhir_endpoints_info_history ADD COLUMN availability DECIMAL(64,4);
+
+CREATE TRIGGER set_timestamp_fhir_endpoint_availability
+BEFORE UPDATE ON fhir_endpoints_availability
+FOR EACH ROW
+EXECUTE PROCEDURE trigger_set_timestamp();
+
+CREATE OR REPLACE FUNCTION populate_existing_tables_availability_info() RETURNS VOID as $$
+    DECLARE 
+        i             record;
+        okay_count      bigint;
+        all_count       bigint;
+    BEGIN
+        FOR i IN SELECT DISTINCT fhir_endpoints_info_history.url FROM fhir_endpoints_info_history
+        LOOP
+            SELECT COUNT(*) INTO all_count FROM fhir_endpoints_info_history WHERE url = i.url AND (operation = 'I' OR operation = 'U');
+            SELECT COUNT(*) INTO okay_count FROM fhir_endpoints_info_history WHERE url = i.url AND http_response = 200 AND (operation = 'I' OR operation = 'U');
+            INSERT INTO fhir_endpoints_availability VALUES (i.url, okay_count, all_count);
+            UPDATE fhir_endpoints_info SET availability = (okay_count * 1.0) / all_count WHERE url = i.url;
```

Comment:
why is `okay_count` multiplied by `1.0` here?

Reply:
in postgres the result of the division is based on the type of the operands. Since okay_count and all_count are both ints I multiplied one by 1.0 to get a decimal result

Reply:
oh okay that makes sense, thanks for the clarification!